Code/Framework/AzTest/AzTest/Platform/iOS/platform_ios_files.cmake

```diff
@@ -9,6 +9,7 @@
 set(FILES
     ../Common/UnixLike/AzTest/ColorizedOutput_UnixLike.cpp
     ../Common/Unimplemented/AzTest/ScopedAutoTempDirectory_Unimplemented.cpp
+    ../Common/Unimplemented/AzTest/Utils_Unimplemented.cpp
```

Comment:
I think iOS could work, but not sure about android.
I turn it off just because we don't have that use case for these two platform unit testing yet.
If you think it is ok to turn it on, I can do so.

Reply:
We do build the Unit Test for those platforms, we don't run them, yet, but we do build them.
So we would this functionality to work on the mobile devices in order for the AWSGameLift test to work.

Reply:
Chiming in to at least confirm the UnixLike implementation will work on Android.

Reply:
Yep, our AR job just build it but not running it for ios and android, so it is hard to verify whether it can work during runtime. (Or it requires developer to test it manually)
Build time should be fine based on doc https://developer.android.com/reference/android/system/Os#setenv(java.lang.String,%20java.lang.String,%20boolean)

I am personally incline to turn them off until AR starts running unit test.

Reply:
That would be for the Java versions of these functions which rely on exceptions being thrown instead of return values for errors.  The native side uses the standard POSIX API linked as reference in those docs so Android will build just fine with the UnixLike file :)

Reply:
phew, cool, still need more inputs should we enable for ios and android. 
As based on AzCore utils, i can see it has android unimplemented but ios enabled. I can follow the same behavior. But it doesn't help that much if there is no pipeline setup to run unit test for these two platforms. (as it is pretty hard to maintain, I don't know what is the current goal for these two platforms, like as developer if I write new unit test, how do I know whether it can work on these two?)

Reply:
> I am personally incline to turn them off until AR starts running unit test.

I agree with the inclination to only turn on what we can adequately test. However, I am a little concerned that if we build these platforms with the `Unimplemented` version, future test writers may not understand that they _can_ use these functions, if they need to. 

Does that sound like a valid concern, or do you all think it's clear enough that "Unimplemented" != "unsupported", and that building these in would be trivial to do & discover if needed later?

Reply:
> how do I know whether it can work on these two?

AzTestRunner can be run on both iOS and Android the same way as on other platforms. There are some helper scripts to run them in https://github.com/o3de/o3de/tree/development/cmake/Tools/Platform. Not all tests will currently pass but a lot of them do I think.

Reply:
I believe helper scripts have to run on windows and mac platform separately? and still need driven manually
I am totally fine to turn it on, just don't like lack of a automation pipeline for testing, and especially this is AzTest util now.

Reply:
Yes. The scripts will always need to be run on a host platform(PC/Mac/Linux) since it needs to be built first and then deployed to device in order to run. The scripts automate the building/deploying/running of the tests. The problem is that we don't have a way to run it as part of AR or nightly since we would need target devices(iOS/Android) connected to the host machines.

Reply:
enabled